hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -0,0 +1,246 @@
+        this.enterprise = BuildInfoProvider.getBuildInfo().isEnterprise();
+    }
+
+    /**
+     * Registers all client statistics and schedules peridic collection of stats
+     */
+    public void start() {
+        if (!enabled) {
+            return;
+        }
+
+        long periodSeconds = properties.getSeconds(PERIOD_SECONDS);
+        if (periodSeconds <= 0) {
+            long defaultValue = Long.parseLong(PERIOD_SECONDS.getDefaultValue());
+            logger.warning("Provided client statistics " + PERIOD_SECONDS.getName()
+                    + " can not be less than or equal to 0. You provided " + periodSeconds
+                    + " seconds as the configuration. Client will use the default value of " + defaultValue + " instead.");
+            periodSeconds = defaultValue;
+        }
+
+        registerStatistics();
+
+        schedulePeriodicStatisticsSendTask(periodSeconds);
+
+        logger.info("Client statistics is enabled with period " + periodSeconds + " seconds.");
+    }
+
+    private ClientConnection getOwnerConnection() {
+        Address ownerConnectionAddress = client.getClientClusterService().getOwnerConnectionAddress();
+        if (null == ownerConnectionAddress) {
+            return null;
+        }
+        ClientConnection connection = (ClientConnection) client.getConnectionManager().getConnection(ownerConnectionAddress);
+        if (null == connection) {
+            return null;
+        }
+
+        int serverVersion = connection.getConnectedServerVersion();
+        if (serverVersion < FEATURE_SUPPORTED_SINCE_VERSION) {
+            if (logger.isFinestEnabled()) {
+                logger.finest(format("Client statistics can not be started since current connected owner server version is less than"
+                        + " the minimum supported server version %s", FEATURE_SUPPORTED_SINCE_VERSION_STRING));
+            }
+            return null;
+        }
+
+        return connection;
+    }
+
+    private void schedulePeriodicStatisticsSendTask(long periodSeconds) {
+        client.getExecutionService().scheduleWithRepetition(new Runnable() {
+            @Override
+            public void run() {
+                ClientConnection ownerConnection = getOwnerConnection();
+                if (null == ownerConnection) {
+                    logger.finest("Can not send client statistics to the server. No owner connection.");
+                    return;
+                }
+
+                final Map<String, String> stats = new HashMap<String, String>();
+
+                stats.put("lastStatisticsCollectionTime", Long.toString(System.currentTimeMillis()));
+
+                stats.putAll(staticStats);
+
+                periodicStats.fillMetrics(stats);
+
+                ThreadPoolExecutor userExecutor = (ThreadPoolExecutor) client.getClientExecutionService().getUserExecutor();
+                stats.put("userExecutor.queueSize", Long.toString(userExecutor.getQueue().size()));
```

Comment:
This should have been exposed as a probe so that also the regular diagnostics can read out this value. THis way the design doesn't get riddles with ad hoc fixes.